cs3api4lab/handlers.py

```diff
@@ -218,7 +231,8 @@ def setup_handlers(web_app, url_path):
         (r"/api/cs3/user", UserInfoHandler),
         (r"/api/cs3/user/claim", UserInfoClaimHandler),
         (r"/api/cs3/user/query", UserQueryHandler),
-        (r"/api/cs3/user/home_dir", HomeDirHandler)
+        (r"/api/cs3/user/home_dir", HomeDirHandler),
+        (r"/api/cs3/locks/create_conflict_file", LockHandler),
```

Comment:
I don't see a good reason to have a dedicated backend api endpoint for this. There's a way to create a copy of a file, so why don't we use it?

From the contents manager docs (for POST):
```
        """Create a new file in the specified path.

        POST creates new files. The server always decides on the name.

        POST /api/contents/path
          New untitled, empty file or directory.
        POST /api/contents/path
          with body {"copy_from" : "/path/to/OtherNotebook.ipynb"}
          New copy of OtherNotebook in path
        """
```

So we can just ask for a copy of original file + current username (which we should have in the frontend as well, right?).

Reply:
Looks like we're missing the copy function in the cm... This is important because the UI also provides that option.

Reply:
I believe the copy function comes from the parent class that we inherit from jupyter_server... But they implement in a similar way (which is getting and saving the file). We could prob make it more efficient, but let's keep the default behaviour.

Reply:
Just for keeping the records: we agreed that a dedicated backend api endpoint can stay.

Rationale for a separate backend endpoint:
- encapsulation of the logic for copyting the file (and recognizing the copy by filename)
- keeping the logic in backend (so it is easier to update in case some edge cases disturb and need action)
- moving the logic to FE is not really simpler -> getting the username to FE (user in the FE – might not be the same as one in Reva, implication of which need further analysis)
- for the future development: if handling some specific scenarios proves that persistence of copy information is necessary, this can be handled here